### PR TITLE
email authentication microservice added

### PR DESCRIPTION
**Pull request by Evgeny-Gromyko**

### CODE REVIEW

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ModsenOnlineStore.EmailAuthentication.Application.Interfaces;
+
+namespace ModsenOnlineStore.EmailAuthentication.Infrastructure.Services
+{
+    public class VerificationCodeGeneratior : IVerificationCodeGeneratior
+    {
+        public string generateCode()
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: use PascalCase for naming

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        {
+            Random random = new Random();
+            string code = "";
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: use StringBuilder

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+            string ch;
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: ch ? what does it mean?

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using ModsenOnlineStore.EmailAuthentication.Application.Interfaces;
+
+namespace ModsenOnlineStore.EmailAuthentication.Infrastructure.Services
+{
+    public class VerificationCodeGeneratior : IVerificationCodeGeneratior
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs:10]: put it into Application layer

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+{
+    public class EmailSendingService : IEmailSendingService
+    {
+        private MailAddress from = new MailAddress("akkdliabreda@gmail.com", "Store Mail Authentication");
+        private NetworkCredential networkCredential = new NetworkCredential("akkdliabreda@gmail.com", "vqshkhtemsppmhha");
+
+        public void SendAuthEmail(string mailAddress = "egrom2002@gmail.com", string title = "", string text = "")
+        {
+            MailAddress to = new MailAddress(mailAddress);
+            MailMessage m = new MailMessage(from, to);
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: to ? m ? and use var keyword

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+using System.Net.Mail;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace ModsenOnlineStore.EmailAuthentication.Infrastructure.Services
+{
+    public class EmailSendingService : IEmailSendingService
+    {
+        private MailAddress from = new MailAddress("akkdliabreda@gmail.com", "Store Mail Authentication");
+        private NetworkCredential networkCredential = new NetworkCredential("akkdliabreda@gmail.com", "vqshkhtemsppmhha");
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: store all these credentials in appsettings.json

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+        public void SendAuthEmail(string mailAddress = "egrom2002@gmail.com", string title = "", string text = "")
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: It is better to name just SendEmail as nothing here about auth

File: ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+
+        [HttpPost]
+        public async Task<IActionResult> RegisterUser(string email)
+        {
+            string code = codeGeneratior.generateCode();
+
+            string title = "Online store verification code";
+
+            string text = $"<h2>Here is your verification code : {code}</h2>" +
+                           "<p>copy it and paste it into the field for entering the verification code</p>";
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs]: store such strings in constants in separate file

File: ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+using System.Net.Mail;
+
+namespace ModsenOnlineStore.EmailAuthentication.API.Controllers
+{
+    [Route("[controller]")]
+    [ApiController]
+    public class EmailAuthenticationController : ControllerBase
+    {
+        private IEmailSendingService emailSendingService;
+        private IVerificationCodeGeneratior codeGeneratior;
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs]: use private readonly for fields
add empty lines between fields, methods etc

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs (hunk 1 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs (hunk 2 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs (hunk 3 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs (hunk 5 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs (hunk 6 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs (hunk 7 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/EmailSendingService.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs (hunk 8 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs (hunk 9 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs]: fixed

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs (hunk 4 above)
Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs:10]: Could you please tell me why it should be stored in the Application layer?

Comment (Evgeny-Gromyko) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs:10]: VerificationCodeGeneratior is the implementation interface IVerificationCodeGeneratior. And as implementation as far as I know it should be stored in the Infrastructure layer.

File: ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs [hunk 10]
```diff
@@ -0,0 +1,35 @@
...
+            this.emailSendingService = emailSendingService;
+            this.codeGeneratior = codeGeneratior;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> RegisterUser(string email)
+        {
+            string code = codeGeneratior.GenerateCode();
+
+            string text = Constants.TextTitle + $"<h2>{code}</h2>" + Constants.TextBody;
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.API/Controllers/EmailAuthenticationController.cs:28]: use string interpolation for full string

File: ModsenOnlineStore.EmailAuthentication.Domain/Constants.cs [hunk 11]
```diff
@@ -0,0 +1,13 @@
+﻿namespace ModsenOnlineStore.EmailAuthentication.Domain
+{
+    public static class Constants
+    {
+
+        public const string Theme = "Online store verification code";
+
+        public const string TextTitle = $"<h2>Here is your verification code</h2>";
+
+        public const string TextBody = "<p>copy it and paste it into " +
```
Comment (elv1sabene) [inline ModsenOnlineStore.EmailAuthentication.Domain/Constants.cs:10]: may be name all of these constants more specifically?

File: ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs (hunk 4 above)
Comment (elv1sabene) [reply ModsenOnlineStore.EmailAuthentication.Infrastructure/Services/VerificationCodeGeneratior.cs:10]: Infrastructure layer should store components that operates with external systems e.g. database. This class just stores logic for generating random strings. But okay, let it be here if it is used only by the mail service.

**Review by elv1sabene — APPROVED** [review]

**Pull request merged**